src/pages/ProfilePage.tsx

```diff
@@ -37,99 +38,287 @@ export const ProfileButton = styled(Button)({
   fontWeight: 500,
 })
 
+interface ProfileData {
+  username: string
+  email: string
+  name: string
+  school: string
+  numAttempts: number
+  pieChartData: PieChartProps
+}
+
 export default function ProfilePage() {
   const [isEditing, setIsEditing] = useState<boolean>(false)
-  const [profileData, setProfileData] = useState<string[]>([
-    'Placeholder',
-    'Placeholder',
-    'Placeholder',
-    'Placeholder',
-  ])
-  const handleEdit = () => {
-    setIsEditing(true)
+
+  const profileDataDefault = {
+    username: 'empty',
+    email: 'empty',
+    name: 'empty',
+    school: 'empty',
+    pieChartData: {
+      numAC: 0,
+      numWA: 0,
+      numTLE: 0,
+      numRTE: 0,
+    },
+    numAttempts: 0,
   }
-  const handleDone = () => {
-    setIsEditing(false)
+  const [profileData, setProfileData] =
+    useState<ProfileData>(profileDataDefault)
+
+  const user = useSelector((state: RootState) => state.user)
+
+  useEffect(() => {
+    fetch(buildPath('/getUserData'), {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ uid: user.id }),
+    })
+      .then((res: Response) => {
+        if (res.ok) {
+          return res.json()
+        }
+        throw Error(res.statusText)
+      })
+      .then((data) => {
+        console.log('Fetched data for ' + data.username)
+        console.log(data)
+
+        const newPieChartData: PieChartProps = {
+          numAC: data.problemsAccepted,
+          numWA: data.problemsWrong,
+          numTLE: data.problemsTLE,
+          numRTE: data.problemsRTE,
+        }
+
+        const newProfileData: ProfileData = {
+          username: 'testuser',
+          email: data.email,
+          name: data.name,
+          school: 'University of Maryland',
+          numAttempts: data.problemsAttempted,
+          pieChartData: newPieChartData,
+        }
+
+        setProfileData(newProfileData)
+      })
+      .catch((error: Error) => {
+        console.log(
+          'Error fetching data for user ' + user.id + ': ' + error.message,
+        )
+      })
+  }, [])
+
+  const onTextFieldChange = (key, newString) => {
+    const jsonCopy: ProfileData = { ...profileData }
+    jsonCopy[key] = newString
+    setProfileData(jsonCopy)
```

Comment:
When the new state depends on the old state, best practice is to use an updater function. See [here](https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state).

So we can do something like:

```
setProfileData(oldData => {
  ...oldData,
  [key]: newString
})
```